hpx/lcos/base_lco_with_value.hpp

```diff
@@ -82,6 +83,17 @@ namespace hpx { namespace lcos
 
         virtual void set_event()
         {
+            set_event_nonvirt(std::is_default_constructible<RemoteResult>());
+        }
+
+        void set_event_nonvirt(std::false_type)
+        {
+            // this shouldn't ever be called
+            HPX_ASSERT(false);
```

Comment:
I think it might just as well be a user error. `base_lco_with_value` can be used in user code as well, so it might be good to have good diagnostics here.